### PR TITLE
Apply separate style to question text if type is group.

### LINKED ISSUE #1450: Subtitle text not behaving as expected
Case 1: The text "Correct!" appears twice, once incorrectly above the question "Cats are better than dogs", and once correctly based on the `enableWhen` element.
```json
{
  "resourceType": "Questionnaire",
  "status": "active",
  "item": [
    {
      "linkId": "2",
      "text": "Demographic Information",
      "type": "group",
      "item": [
        {
          "linkId": "2.1",
          "text": "Cats are better than dogs",
          "type": "boolean"
        },
        {
          "linkId": "2.1.1",
          "text": "Correct!",
          "type": "display",
          "enableWhen": [
            {
              "question": "2.1",
              "operator": "=",
              "answerBoolean": true
            }
          ]
        }
      ]
    }
  ]
}
```
image

Case 2: When `enableWhen` is a child of the item, the "Correct!" text never appears. It still shows up incorrectly above the question, like in Case 1.
```json
{
  "resourceType": "Questionnaire",
  "status": "active",
  "item": [
    {
      "linkId": "2",
      "text": "Demographic Information",
      "type": "group",
      "item": [
        {
          "linkId": "2.1",
          "text": "Cats are better than dogs",
          "type": "boolean",
          "item": [
            {
              "linkId": "2.1.1",
              "text": "Correct!",
              "type": "display",
              "enableWhen": [
                {
                  "question": "2.1",
                  "operator": "=",
                  "answerBoolean": true
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
```
image

This is using the published version of `com.google.android.fhir:data-capture:0.1.0-beta03`.

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1450

**Description**
Earlier, it was okay to have single attribute resource to set style to question text in header view in all views, but now requirement is to set different style to question text if type is group. Therefore we can not keep same existing files and set attribute resource at run time in kolin code according to the group type, as there is no such api which which accepts attribute resource.
So created separate xml resource file for group type header view and updated changes accordingly.

**Alternative(s) considered**
Another approach is to set attribute resource to question text if type is group in header view in kotlin code, but there is no such api which accept attribute as resource to set the style.

**Type**
Choose one: Feature 
**Screenshots (if applicable)**
Attaching screenshot for demo purpose only.
@shelaghm 
Screen Shot 2022-08-09 at 4 28 55 PM

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt [hunk 1]
```diff
@@ -41,42 +43,31 @@ internal class QuestionnaireItemHeaderView(context: Context, attrs: AttributeSet
...
+    if (localizedText.isNullOrEmpty()) {
+      GONE
+    } else {
+      VISIBLE
+    }
+}
+
+/** Returns [VISIBLE] if any of the [view] is visible, else returns [GONE]. */
+internal fun isViewVisible(vararg view: TextView): Int {
+  view.forEach {
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt:67]: use `any`

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt [hunk 2]
```diff
@@ -41,42 +43,31 @@ internal class QuestionnaireItemHeaderView(context: Context, attrs: AttributeSet
...
-      ) {
-        VISIBLE
-      } else {
-        GONE
-      }
+    visibility = isViewVisible(prefix, question, hint)
+  }
+}
+
+internal fun TextView.updateText(localizedText: Spanned?) {
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt:55]:
```suggestion
internal fun TextView.updateTextAndVisibility(localizedText: Spanned?) {
```

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt [hunk 3]
```diff
@@ -41,42 +43,29 @@ internal class QuestionnaireItemHeaderView(context: Context, attrs: AttributeSet
...
+  visibility =
+    if (localizedText.isNullOrEmpty()) {
+      GONE
+    } else {
+      VISIBLE
+    }
+}
+
+/** Returns [VISIBLE] if any of the [view] is visible, else returns [GONE]. */
+internal fun isViewVisible(vararg view: TextView): Int {
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt:66]: we discussed that this function naming is incorrect!

it needs to return a boolean.

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt (hunk 3 above)
Comment (santosh-pingle) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt:66]: We also discussed one of the option is rename it to getVisibility and then keep return type Int, can you please confirm?

Comment (santosh-pingle) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt]: renamed it to the getVisibility()

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt [hunk 4]
```diff
@@ -41,42 +43,29 @@ internal class QuestionnaireItemHeaderView(context: Context, attrs: AttributeSet
...
+  visibility =
+    if (localizedText.isNullOrEmpty()) {
+      GONE
+    } else {
+      VISIBLE
+    }
+}
+
+/** Returns [VISIBLE] if any of the [view] is visible, else returns [GONE]. */
+internal fun getVisibility(vararg view: TextView): Int {
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt:66]:
```suggestion
internal fun getViewGroupVisibility(vararg view: TextView): Int {
```

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**